crates/pallet-domains/src/lib.rs

```diff
@@ -584,6 +586,29 @@ impl<T: Config> Pallet<T> {
             }
         }
 
+        let bundle_created_on_valid_primary_block =
+            match pallet_receipts::PrimaryBlockHash::<T>::get(
+                DomainId::SYSTEM,
+                bundle.header.primary_number,
+            ) {
+                Some(block_hash) => block_hash == bundle.header.primary_hash,
+                // The `initialize_block` of non-system pallets is skipped in the `validate_transaction`,
+                // thus the hash of best block, which is recorded in the this pallet's `on_initialize` hook,
+                // is unavailable in pallet-receipts at this point.
+                None => frame_system::Pallet::<T>::parent_hash() == bundle.header.primary_hash,
+            };
+
+        if !bundle_created_on_valid_primary_block {
+            log::error!(
+                target: "runtime::domains",
+                "Bundle is probabaly created on a primary fork #{:?}, expected: {:?}, got: {:?}",
```

Comment:
```suggestion
                "Bundle is probably created on a primary fork #{:?}, expected: {:?}, got: {:?}",
```

It is also possible that the bundle points to a future block that this node doesn't sync yet.

Reply:
It's indeed possible, but unlike the receipts, we probably don't want this optimization as we'd like the fork-aware bundles to be strongly ensured.